apps/src/templates/SchoolInfoConfirmationDialog.jsx

```diff
@@ -16,41 +16,88 @@ class SchoolInfoConfirmationDialog extends Component {
     schoolName: PropTypes.string,
     onUpdate: PropTypes.func,
     onConfirm: PropTypes.func,
-    isOpen: PropTypes.bool.isRequired
+    scriptData: PropTypes.object,
+    onClose: PropTypes.func,
+    isOpen: PropTypes.bool
   };
 
   constructor(props) {
     super(props);
-
     this.state = {
       showSchoolInterstitial: false,
       schoolName: props.schoolName,
       isOpen: true
     };
   }
 
-  handleUpdateClick = () => {
+  static defaultProps = {
+    schoolName: '',
+    onUpdate: () => {},
+    onConfirm: () => {},
+    scriptData: {},
+    onClose: () => {},
+    isOpen: true
+  };
+
+  componentDidMount() {
+    const {schoolName} = this.state;
+    if (!schoolName && schoolName.length > 0) {
+      fetch('/dashboardapi/v1/users/me/school_name')
```

Comment:
as it is currently, this request is hitting the incorrect API endpoint. the one you've set up in `routes.rb` points to `/api/v1/users/:user_id/school_name`. to look at the route that is created from an entry in the routes file, you can use the `rake routes` command